pyproject.toml

```diff
@@ -197,11 +160,47 @@ ci = [
 ]
 
 [tool.pixi.dependencies]
-gdal = "==3.12.1"
-psycopg2-binary = ">=2.9.10,<3"
+aiohttp = ">3"
+arelle-release = ">=2.38.0,<3"
+beautifulsoup4 = ">=4.14.3,<5"
+# Install PUDL as a conda package from git:
+"catalystcoop.pudl" = { git = "https://github.com/catalyst-cooperative/pudl.git", branch = "pixi-setup" }
+# Install PUDL from a local directory for convenience during development:
+# "catalystcoop.pudl" = { path = "../pudl" }
+cfgv = ">=3.5.0,<4"
+coloredlogs = ">=14.0,<15"
+coverage = ">=7.13.1,<8"
+dask = ">=2025.12.0,<2026"
+deltalake = ">=1.2.1,<2"
+doc8 = ">=2.0.0,<3"
+feedparser = ">=6.0.12,<7"
+frictionless = ">=5.18.1,<6"
 numba = ">=0.63.1,<0.64"
-pandas = ">=2.3.3,<3"
 numpy = ">=2.3.5,<3"
+pandas = ">=2.3.3,<3"
+pre-commit = ">=4.5.1,<5"
+psycopg2-binary = ">=2.9.10,<3"
+pyarrow = ">=20"
+pydantic = ">=2.12.5,<3"
+pydocstyle = ">=6.3.0,<7"
+pytest = ">=9.0.2,<10"
+pytest-asyncio = ">=1.3.0,<2"
+pytest-console-scripts = ">=1.4.1,<2"
+pytest-cov = ">=7.0.0,<8"
+pytest-mock = ">=3.15.1,<4"
+python-dotenv = ">=1.2.1,<2"
+pyyaml = ">=6.0.3,<7"
+ruff = ">=0.14.10,<0.15"
+sphinx = ">=9.0.4,<10"
+sphinx-autoapi = ">=3.6.1,<4"
+sphinx-issues = ">=5.0.1,<6"
+tqdm = ">=4.67.1,<5"
+types-requests = ">=2.32.4.20250913,<3"
 
 [tool.pixi.activation]
 scripts = ["scripts/check_playwright_installation.sh"]
+
+[tool.pixi.activation.env]
+# Prevents every single pixi run/install invocation from updating the lockfile.
```

Comment:
Kind of surprising default behavior (and not what [the docs](https://pixi.prefix.dev/latest/workspace/lockfile/#how-to-use-a-lock-file) imply would happen: "Running the following commands will check and automatically update the lock file if you changed any dependencies") but I'm glad you were able to work around it!